src/Controllers/TaxonomyDirectoryController.php

```diff
@@ -29,7 +35,9 @@ public function index(HTTPRequest $request)
     {
         $termString = $request->param('ID');
 
-        $pages = Page::get()->filter(['Terms.Name' => $termString]);
+        $field = $this->config()->get('lookup_field');
+        $term = TaxonomyTerm::get()->filter([$field => $termString])->first();
+        $pages = $term ? $term->Pages() : [];
```

Comment:
This relies on the inverse relationship `Pages` being defined on `TaxonomyTerm`. In the case of CWP this is defined by the `cwp/cwp` module:

https://github.com/silverstripe/cwp/blob/f47a2225d8b52f81f8038767e8e82be7764f4366/src/Extensions/TaxonomyTermExtension.php#L15-L17

`cwp/cwp` is not a specific requirement of this module so this line just errors without that module. You might have to look into the alternative solution you were postulating earlier.